homeassistant/components/honeywell/__init__.py

```diff
@@ -115,10 +118,18 @@ def get_somecomfort_client(username, password):
 class HoneywellData:
     """Get the latest data and update."""
 
-    def __init__(self, hass, config, client, username, password, devices):
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        client: somecomfort.SomeComfort,
+        username: Any,
+        password: Any,
+        devices: dict[Any, somecomfort.Device],
```

Comment:
I am guessing that `device.deviceid` are also strings. So this probably also needs to be tweaked:
```suggestion
        devices: dict[str, somecomfort.Device],
```

Reply:
Done.